resource-managers/kubernetes/core/src/test/scala/org/apache/spark/deploy/kubernetes/submit/ClientV2Suite.scala

```diff
@@ -169,30 +189,85 @@ class ClientV2Suite extends SparkFunSuite with BeforeAndAfter {
             .endMetadata()
         }
       })
-    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver())
+    when(initContainerComponentsProvider.provideContainerLocalizedFilesResolver(any[String]))
       .thenReturn(containerLocalizedFilesResolver)
+    when(initContainerComponentsProvider.provideDriverPodFileMounter())
+        .thenReturn(fileMounter)
     when(submittedDependenciesSecretBuilder.build())
       .thenReturn(INIT_CONTAINER_SECRET)
     when(kubernetesClient.pods()).thenReturn(podOps)
     when(podOps.create(any())).thenAnswer(new Answer[Pod] {
       override def answer(invocation: InvocationOnMock): Pod = {
         new PodBuilder(invocation.getArgumentAt(0, classOf[Pod]))
           .editMetadata()
-          .withUid(DRIVER_POD_UID)
+            .withUid(DRIVER_POD_UID)
           .endMetadata()
-          .withKind(DRIVER_POD_KIND)
+            .withKind(DRIVER_POD_KIND)
           .withApiVersion(DRIVER_POD_API_VERSION)
           .build()
       }
     })
     when(podOps.withName(s"$APP_RESOURCE_PREFIX-driver")).thenReturn(namedPodResource)
+    when(pythonSubmissionResources.sparkJars).thenReturn(Seq.empty[String])
+    when(pythonSubmissionResources.primarySparkResource(any()))
+      .thenReturn(RESOLVED_PYSPARK_PRIMARY_FILE)
+    when(pythonSubmissionResources.pySparkFiles).thenReturn(PYSPARK_FILES.toArray)
+    when(pythonSubmissionResources.arguments).thenReturn(Array(PYSPARK_FILES.mkString(","), "500"))
+    when(pythonSubmissionResources.driverPod(
+      any[DriverInitContainerComponentsProvider],
+      mockitoEq(RESOLVED_PYSPARK_PRIMARY_FILE),
+      mockitoEq(RESOLVED_PYSPARK_FILES.mkString(",")),
+      any[String],
+      any[PodBuilder])).thenAnswer( new Answer[Pod] {
+        override def answer(invocation: InvocationOnMock) : Pod = {
+          invocation.getArgumentAt(0, classOf[DriverInitContainerComponentsProvider])
```

Comment:
It's possible to use mocks to stub classes, but it's certainly not preferred. Mocks also can't be used to stub final methods or final classes so if we're mocking classes we have to make an assumption that we aren't doing those things - best to avoid the uncertainty entirely.